### PR TITLE
include-terminated option in Deployment list

### PR DESCRIPTION
**Pull request by kdh0102**
`show-all` option does not work due to server-side changes.
I have changed to pass `archived` option to the server.

Also, I modified the option name to `include-terminated`. Since we have `limit` option, using both `show-all` and `limit` options does not seem to be intuitive.

### CODE REVIEW

File: pfcli/service/client/deployment.py [hunk 1]
```diff
@@ -28,9 +28,9 @@ def scale_deployment(self, deployment_id: str, replicas: int) -> None:
             err_prefix=f"Deployment ({deployment_id}) cannot be scaled.",
         )(pk=deployment_id, path=f"scale", data=str(replicas))
 
-    def list_deployments(self, project_id: str) -> Dict[str, Any]:
+    def list_deployments(self, project_id: str, archived: Optional[bool]) -> Dict[str, Any]:
```
Comment (pigbug419) [inline pfcli/service/client/deployment.py:31]:
```suggestion
    def list_deployments(self, project_id: str, archived: Optional[bool] = None) -> Dict[str, Any]:
```
You want to send also `None` value to server?

File: pfcli/deployment.py [hunk 2]
```diff
@@ -154,10 +154,9 @@
 
 @app.command()
 def list(
-    show_all: bool = typer.Option(
+    include_terminated: bool = typer.Option(
         False,
-        "--all",
-        "-a",
+        "--include-terminated",
```
Comment (pigbug419) [inline pfcli/deployment.py:159]: Do we need to support list of only archived deployments?

**Review by pigbug419 — CHANGES_REQUESTED** [review]
Thank you, I've left some comments.

File: pfcli/service/client/deployment.py (hunk 1 above)
Comment (kdh0102) [reply pfcli/service/client/deployment.py]: I found out sending `None` is treated as sending `False`.
So, I changed the type from Optional[bool] to bool.

File: pfcli/deployment.py (hunk 2 above)
Comment (kdh0102) [reply pfcli/deployment.py:159]: I do not think so. Maybe later we can support list by deployment status.

**Comment by pigbug419** [comment]
@kdh0102 Could you revolve this conflicts?

**Comment by kdh0102** [comment]
@pigbug419 Done

**Review by pigbug419 — APPROVED** [review]
LGTM!

**Pull request merged**